tests/php/test-class-customize-snapshot-manager.php

```diff
@@ -171,6 +184,183 @@ function test_create_post_type() {
 	}
 
 	/**
+	 * @see Customize_Snapshot_Manager::filter_theme_mod_nav_menu_locations()
+	 */
+	function test_filter_theme_mod_nav_menu_locations() {
+		wp_set_current_user( $this->user_id );
+		$_POST = wp_slash( array(
+			'nonce' => wp_create_nonce( 'save-customize_' . $this->wp_customize->get_stylesheet() ),
+			'snapshot_uuid' => self::UUID,
+			'snapshot_customized' => '{"nav_menu_locations[foobar]":{"value":"' . self::MENU_ID . '"}}',
+		) );
+		add_action( 'after_setup_theme', function() {
+			register_nav_menu( 'foobar', 'Custom Menu' );
+		} );
+		$this->do_customize_boot_actions( true );
+
+		$manager = new Customize_Snapshot_Manager( $this->plugin );
+		$manager->set_snapshot_uuid();
+		$manager->save_snapshot();
+		$manager->snapshot()->is_preview = true;
+
+		$menu_locations = $manager->filter_theme_mod_nav_menu_locations( array( 'foobar' => '' ) );
+		$this->assertEquals( self::MENU_ID, $menu_locations['foobar'] );
+	}
+
+	/**
+	 * @see Customize_Snapshot_Manager::filter_wp_get_nav_menus()
+	 */
+	function test_filter_wp_get_nav_menus() {
+
+		$_POST = wp_slash( array(
+			'nonce' => wp_create_nonce( 'save-customize_' . $this->wp_customize->get_stylesheet() ),
+			'snapshot_uuid' => self::UUID,
+			'snapshot_customized' => '{"nav_menu[' . self::MENU_ID . ']":{"value":{}}}',
+		) );
+
+		wp_set_current_user( $this->user_id );
+		$this->do_customize_boot_actions( true );
+
+		$manager = new Customize_Snapshot_Manager( $this->plugin );
+		$manager->set_snapshot_uuid();
+		$manager->save_snapshot();
+		$manager->snapshot()->is_preview = true;
+
+		$menus = array();
+		$menus[] = $this->manager->get_nav_menu_object( self::MENU_ID, array() );
+		$this->assertEquals( array_values( $menus ), $manager->filter_wp_get_nav_menus( array() ) );
+	}
+
+	/**
+	 * @see Customize_Snapshot_Manager::filter_wp_get_nav_menu_items()
+	 */
+	function test_filter_wp_get_nav_menu_items() {
+
+		$this->wp_customize->add_setting( 'nav_menu_item[' . self::MENU_ID . ']');
+
+		$_POST = wp_slash( array(
+			'nonce' => wp_create_nonce( 'save-customize_' . $this->wp_customize->get_stylesheet() ),
+			'snapshot_uuid' => self::UUID,
+			'snapshot_customized' => '{"nav_menu_item[' . self::MENU_ID . ']":{"value":{"nav_menu_term_id":"' .
+				self::MENU_ID . '","status":"publish","position":"1","title":"","post_id":"' . self::MENU_ID . '"}}}',
+		) );
+
+		wp_set_current_user( $this->user_id );
+		$this->do_customize_boot_actions( true );
+
+		$manager = new Customize_Snapshot_Manager( $this->plugin );
+		$manager->set_snapshot_uuid();
+		$manager->save_snapshot();
+		$manager->snapshot()->is_preview = true;
+
+		$menu_items = array();
+		$menu_items[] = $this->manager->value_as_wp_post_nav_menu_item(
```

Comment:
I think we can avoid duplicating the `WP_Customize_Nav_Menu_Item_Setting::value_as_wp_post_nav_menu_item()` method here by doing:

```php
$setting_id = sprintf( 'nav_menu_item[%d]', static::MENU_ID );
$nav_menu_item = new WP_Customize_Nav_Menu_Item_Setting( $this->customize_manager, $setting_id, array(
    value: array( /* ... */ )
) );
$menu_items[] = $nav_menu_item-> value_as_wp_post_nav_menu_item();
```